src/test/test-create-user.ts

```diff
@@ -0,0 +1,84 @@
+import { expect } from 'chai';
+import { createdUser, createRepUser, queryBase } from './input';
+import { createToken } from '../create-token';
+import { findUser } from '../find-user';
+import crypto from 'node:crypto';
+import { promisify } from 'node:util';
+
+describe('Testing createUser', () => {
+  it('should create an user and return it', async () => {
+    const token = createToken(0, true);
+    const input = {
+      name: 'eu',
+      email: 'eu@gmail.com',
+      birthDate: '27/12/1900',
+      password: 'mudar123',
+    };
+    const response = await queryBase(createdUser, input, token);
+    const user = await findUser(input.email);
+    expect(user.name).to.be.eq(input.name);
+    expect(user.birthDate).to.be.eq(input.birthDate);
+    const promiseCrypto = promisify(crypto.scrypt);
+    const derivedKey = (await promiseCrypto(input.password, 'salt', 10)) as Buffer;
+    const hash = derivedKey.toString('hex');
+    expect(user.hash).to.be.eq(hash);
+    const id = `${user.id}`;
+    expect(response.data.data.createUser).to.be.deep.eq({
+      birthDate: user.birthDate,
+      email: user.email,
+      id,
+      name: input.name,
+    });
```

Comment:
```suggestion
    expect(response.data.data.createUser).to.be.deep.eq({
      birthDate: user.birthDate,
      email: user.email,
      id: user.id,
      name: input.name,
    });
```

Algumas vairáveis acabam complicando mais do que usar o valor direto né?

Reply:
então, é que se eu coloco user.id retorna um ID, só que o graphQL retorna em uma string ai por isso eu acabo tendo que usar a variável